common/menu.c

```diff
@@ -800,16 +800,42 @@ void menuLoop(void) {
         int entries_count = layoutobj->posEnd[0];
         layoutobj = &themeCurrent.layoutObjects[ThemeLayoutId_MenuList];
 
-        if (menu->nEntries > entries_count) {
-            int wanted_x = clamp(-menu->curEntry * layoutobj->posEnd[0], -(menu->nEntries - entries_count) * layoutobj->posEnd[0], 0);
-            menu->xPos += v;
-            v += (wanted_x - menu->xPos) / 3;
-            v /= 2;
+        // Gentle Realign only when not manually moving
+        if (menu->slideSpeed == 0) {
+            if (menu->nEntries > entries_count) {
+                int wanted_x = clamp(-menu->curEntry * layoutobj->posEnd[0], -(menu->nEntries - entries_count) * layoutobj->posEnd[0], 0);
+                menu->xPos += v;
+                v += (wanted_x - menu->xPos) / 3;
+                v /= 2;
+            }
+            else {
+                menu->xPos = v = 0;
+            }
         }
         else {
-            menu->xPos = v = 0;
+            menu->xPos += menu->slideSpeed;
+
+            if (abs(menu->slideSpeed) > 2) {
```

Comment:
Should be `absf` instead, otherwise it's using the double variant instead of the correct float variant.

Reply:
You sure? slideSpeed is an int in this usage.

Reply:
Ah, nvm then.